lib/riscv32_table.h

```diff
@@ -0,0 +1,330 @@
+/* riscv32_table.h --
+ * Copyright 2024 Rivos Inc.
+ * All Rights Reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors:
+ *      David Abdurachmanov <davidlt@rivosinc.com>
+ */
+
+_S(0, "io_setup")
+_S(1, "io_destroy")
+_S(2, "io_submit")
+_S(3, "io_cancel")
+_S(5, "setxattr")
+_S(6, "lsetxattr")
+_S(7, "fsetxattr")
+_S(8, "getxattr")
+_S(9, "lgetxattr")
+_S(10, "fgetxattr")
+_S(11, "listxattr")
+_S(12, "llistxattr")
+_S(13, "flistxattr")
+_S(14, "removexattr")
+_S(15, "lremovexattr")
+_S(16, "fremovexattr")
+_S(17, "getcwd")
+_S(18, "lookup_dcookie")
+_S(19, "eventfd2")
+_S(20, "epoll_create1")
+_S(21, "epoll_ctl")
+_S(22, "epoll_pwait")
+_S(23, "dup")
+_S(24, "dup3")
+_S(25, "fcntl64")
+_S(26, "inotify_init1")
+_S(27, "inotify_add_watch")
+_S(28, "inotify_rm_watch")
+_S(29, "ioctl")
+_S(30, "ioprio_set")
+_S(31, "ioprio_get")
+_S(32, "flock")
+_S(33, "mknodat")
+_S(34, "mkdirat")
+_S(35, "unlinkat")
+_S(36, "symlinkat")
+_S(37, "linkat")
+_S(39, "umount2")
+_S(40, "mount")
+_S(41, "pivot_root")
+_S(42, "nfsservctl")
```

Comment:
I am not sure about this one. Entry:
`42	common	nfsservctl			sys_ni_syscall`
It exist, but points to `sys_ni_syscall` which is `-ENOSYS`. Otherwise I remember this being a legacy one (dead), but still incl. in other arches.

Reply:
Looking at the manpage in the history section:
`Removed in Linux 3.1.  Removed in glibc 2.28.`
Maybe is the reason.

Reply:
Correct. It's an old stuff from x86 days IIRC, but it's on pretty much every arch in audit:
```
lib/aarch64_table.h:_S(42, "nfsservctl")
lib/arm_table.h:_S(169, "nfsservctl")
lib/i386_table.h:_S(169, "nfsservctl")
lib/ppc_table.h:_S(168, "nfsservctl")
lib/s390_table.h:_S(169, "nfsservctl")
lib/s390x_table.h:_S(169, "nfsservctl")
lib/x86_64_table.h:_S(180, "nfsservctl")
```
This was one reason I left it. It's still in the table, but it points to `sys_ni_syscall` which basically returns that this syscall is not implemented.

Reply:
I don't think that leaving it there would make any problems, just wanted to discuss so we have it covered :)